cmd/client.go

```diff
@@ -242,6 +252,129 @@ type Config struct {
 	UploadLimit       int64
 	DownloadLimit     int64
 	Transport         http.RoundTripper
+	// Force TLS
+	forceTLS bool
+}
+
+// getCredsChain returns an []credentials.Provider array for the config
+// and the STS configuration (if present)
+func (config *Config) getCredsChain() ([]credentials.Provider, *probe.Error) {
```

Comment:
Not sure if it really adds anything to move the existing function inside the `struct`. It's pretty clean to keep `Config` just a struct that holds data and doesn't provide functionality. It also makes the PR more complex to review (more changes).

It's not a deal-breaker, but I think it was fine the way it was.

Reply:
I agreed that it's better to split the functions. Moving them into `Config` is probably personal preference.